packages/calcite-components/src/tests/commonTests/utils.ts

```diff
@@ -74,3 +74,26 @@ export function getTagOrHTMLWithBeforeContent(componentTestSetup: TagOrHTML | Ta
 export function propToAttr(name: string): string {
   return name.replace(/([a-z0-9])([A-Z])/g, "$1-$2").toLowerCase();
 }
+
+export function validateCaretIndex({
+  page,
+  componentTag,
+  nestedInputTypeSelector = "input",
+  position,
+}: {
+  page: E2EPage;
+  componentTag: string;
+  nestedInputTypeSelector?: "textarea" | "input";
```

Comment:
Nitpick: for consistency, should we rename this to `shadowInputTypeSelector` or something similar?